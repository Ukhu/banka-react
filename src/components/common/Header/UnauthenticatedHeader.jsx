import React from 'react';
import { Link } from 'react-router-dom';

const UnauthenticatedHeader = () => (
  <header>
    <div className="container">
      <div id="header-title">
        <Link to="/">
          <h1>Banka</h1>
        </Link>
      </div>
      <div id="header-links">
        <p className="login-signup">
          <Link className="pre-login-link" to="/signup">
            Sign Up
          </Link>
        </p>
        <p className="login-signup">
            Sign In
        </p>
      </div>
    </div>
  </header>
);

export default UnauthenticatedHeader;
