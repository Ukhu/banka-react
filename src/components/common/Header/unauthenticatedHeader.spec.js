import React from 'react';
import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import UnauthenticatedHeader from './UnauthenticatedHeader';

configure({ adapter: new Adapter() });

describe('Unauthenticated Header component <Unauthenticated />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <UnauthenticatedHeader />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('has the header tag', () => {
    const wrapper = shallow(
      <UnauthenticatedHeader />,
    );

    expect(wrapper.find('header')).toHaveLength(1);
  });

  it('has 3 links', () => {
    const wrapper = shallow(
      <UnauthenticatedHeader />,
    );

    expect(wrapper.find('p')).toHaveLength(2);
    expect(wrapper.find('h1')).toHaveLength(1);
  });
});
