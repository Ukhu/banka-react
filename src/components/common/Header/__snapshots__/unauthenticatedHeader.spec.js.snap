// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthenticated Header component <Unauthenticated /> renders correctly 1`] = `ShallowWrapper {}`;
